reV/hybrids/hybrids.py

```diff
@@ -0,0 +1,1170 @@
+            meta. Note that column names will likely have to be prefixed
+            with "solar_" or "wind_". By default None.
+        limits : dict, optional
+            Option to specify mapping (in the form of a dictionary) of
+            {colum_name: max_value} representing the upper limit (maximum
+            value) for the values of a column in the merged meta. For example,
+            `limits={'solar_capacity': 100}` would limit all the values of the
+            solar capacity in the merged meta to a maximum value of 100.
+            This limit is applied *BEFORE* ratio calculations. The names of
+            the columns should match the column names in the merged meta, so
+            they are likely prefixed with one of the prefixes defined at the
+            top of this module (SOLAR_PREFIX or WIND_PREFIX). By default,
+            None (no limits applied).
+        ratios : dict, optional
+            Option to set a single ratio or ratio bounds on pairs of columns
+            in the merged meta. A detailed explanation follows.
+            This input is a mapping (dictionary), and should have the form
+            {`RatioColumns`: ratio}. The key is an instance of the
+            RatioColumns tuple, which specifies the two column names used
+            in the ratio calculation and optionally specifies if a column name
+            is fixed in the ratio calculation (the "fixed" input should always
+            match the value of either "num" or "denom"). The names of the
+            columns should match the column names in the merged meta, so they
+            must be prefixed with one of the prefixes defined at the top of
+            this module (SOLAR_PREFIX or WIND_PREFIX). The value pointed to
+            by this key specifies the desired ratio or tre ratio bounds. The
+            ratio can be a single value or an iterable of length two (e.g. a
+            two-tuple) representing the lower and upper ratio bounds.
+            A single value is treated as both an upper and a lower bound.
+            For example,
+            `ratios={RatioColumns('solar_capacity', 'wind_capacity'): 1}`
+            would adjust both the solar and wind capacities to always be equal.
+            On the other hand,
+            `ratios={RatioColumns('solar_capacity', 'wind_capacity'):
+            (0.5, 1.5)}` would adjust both the solar and wind capacities
+            such that the solar and wind capacity ratio is always between half
+            and double (e.g., no capacity value would be more than double the
+            other). Finally,
+            `ratios={RatioColumns('solar_capacity', 'wind_capacity',
+            fixed='wind_capacity'): (0.5, 1.5)}` would limit the solar and
+            wind capacities to be between half and double **without altering
+            the value of the wind_capacity** (e.g. only the solar capacity
+            would be adjusted). By default, None (no limit).
```

Comment:
A few comments:

1. Something this complicated may deserve its own class. In argument docstrings for `ratios` you could just say "Option to set ratios... see the HybridsRatio class for details". You could have the class be a general HybridLimits class that does the logic for both limits and ratio limits. 
2. Having different specification between the CLI and the python class is going to be very difficult to maintain (looks like you've already spent quite a lot of time writing the separate doc strings). I would recommend defining a single format that is CLI-friendly.
3. This is a dictionary but i expect (hope) that you cannot set ratio bounds on more than a single pair of columns? I feel like the logic would get pretty messy with ratio bounds on capacity AND something else (no idea why you would want to do this anyway). 
4. Possible options to resolve bullets 2 and 3 above could be argument format like: `[[ratio_col_numerator, ratio_col_denominator], [ratio1, (optional)ratio2]]` or two arguments like: `ratio_cols=[ratio_col_numerator, ratio_col_denominator], ratio_bounds=(0, 1)` I actually think the 2x arguments would be clean and would be more cli-friendly in the long run even if you do have to add another argument.

Reply:
After conversation: fixed is unnecessary, you can set bounds that will build full capacity of either wind or solar. One good option for args is `ratio_cols="wind_col/solar_col", ratio_bounds=(bound1, bound2)` and then just assert a single `/` char is in the ratio_cols string and do the min/max on ratio_bounds to get the ratio min/max bounds

Reply:
Implemented refactor as mentioned in second comment. Used `ratio` instead of `ratio_cols`.